libbuteosyncfw/profile/SyncSchedule.cpp

```diff
@@ -558,45 +559,79 @@ bool SyncSchedule::isSyncScheduled(const QDateTime &aActualDateTime, const QDate
     return reference.secsTo(aActualDateTime) > (interval * 60);
 }
 
-DaySet SyncSchedulePrivate::parseDays(const QString &aDays) const
+SyncSchedule::Days SyncSchedulePrivate::parseDays(const QString &aDays) const
 {
-    DaySet daySet;
+    SyncSchedule::Days days = 0;
+    const SyncSchedule::Day fromStrDays[] =
+        {SyncSchedule::Monday, SyncSchedule::Tuesday,
+         SyncSchedule::Wednesday, SyncSchedule::Thursday,
+         SyncSchedule::Friday, SyncSchedule::Saturday,
+         SyncSchedule::Sunday};
     if (!aDays.isNull()) {
         QStringList dayList = aDays.split(DAY_SEPARATOR,
                                           QString::SkipEmptyParts);
         foreach (QString dayStr, dayList) {
             bool ok;
             int dayNum = dayStr.toInt(&ok);
-            if (ok) {
-                daySet.insert(dayNum);
+            if (ok && dayNum > 0 && dayNum <= 7) {
+                days |= fromStrDays[dayNum - 1];
             }
         }
     }
 
-    return daySet;
+    return days;
 }
 
-QString SyncSchedulePrivate::createDays(const DaySet &aDays) const
+QString SyncSchedulePrivate::createDays(const SyncSchedule::Days &aDays) const
 {
     QStringList dayList;
 
-    foreach (int dayNum, aDays) {
-        dayList.append(QString::number(dayNum));
+    if (aDays & SyncSchedule::Monday) {
+        dayList.append(QString::fromLatin1("1"));
+    }
+    if (aDays & SyncSchedule::Tuesday) {
+        dayList.append(QString::fromLatin1("2"));
+    }
+    if (aDays & SyncSchedule::Wednesday) {
+        dayList.append(QString::fromLatin1("3"));
+    }
+    if (aDays & SyncSchedule::Thursday) {
+        dayList.append(QString::fromLatin1("4"));
+    }
+    if (aDays & SyncSchedule::Friday) {
+        dayList.append(QString::fromLatin1("5"));
+    }
+    if (aDays & SyncSchedule::Saturday) {
+        dayList.append(QString::fromLatin1("6"));
+    }
+    if (aDays & SyncSchedule::Sunday) {
+        dayList.append(QString::fromLatin1("7"));
     }
 
     return dayList.join(DAY_SEPARATOR);
 }
 
-bool SyncSchedulePrivate::adjustDate(QDateTime &aTime, const DaySet &aDays) const
+bool SyncSchedulePrivate::daysMatch(SyncSchedule::Days aDays, int aDay)
+{
+    const SyncSchedule::Day fromQtDays[] =
+        {SyncSchedule::Monday, SyncSchedule::Tuesday,
+         SyncSchedule::Wednesday, SyncSchedule::Thursday,
+         SyncSchedule::Friday, SyncSchedule::Saturday,
+         SyncSchedule::Sunday};
+
+    return aDay >= Qt::Monday && aDay <= Qt::Sunday && (aDays & fromQtDays[aDay - 1]);
+}
+
+bool SyncSchedulePrivate::adjustDate(QDateTime &aTime, const SyncSchedule::Days &aDays) const
```

Comment:
Minor: the latter could be passed by-value as it's just an int now.

Reply:
Good point, I've corrected here and in the other places.